mod/ckeditor/views/default/ckeditor/init.php

```diff
@@ -1,15 +1,32 @@
 <?php
 /**
  * Initialize the CKEditor script
- * 
+ *
  * Doing this inline enables the editor to initialize textareas loaded through ajax
  */
+
+// someone does not want the editor enabled for this field
 if (!elgg_extract('editor', $vars, true)) {
 	return;
 }
+
+$id = elgg_extract('id', $vars);
+if (!$id) {
+	return;
+}
+
+$editor_type = elgg_extract('editor_type', $vars); // eg simple
+
+
+// editor_type
+$config = 'elgg/ckeditor/config';
+if ($editor_type && elgg_view_exists("elgg/ckeditor/config/{$editor_type}.js")) {
+	$config = "elgg/ckeditor/config/{$editor_type}";
+}
+
 ?>
 <script>
-	require(['elgg/ckeditor'], function (elggCKEditor) {
-		elggCKEditor.bind('.elgg-input-longtext');
+	require(['elgg/ckeditor', '<?php echo $config; ?>'], function (elggCKEditor, config) {
```

Comment:
How would you suggest not breaking the `elgg/ckeditor/config` and allow changing config? If would like to switch in full configs based on the data attribute... So not just adjusting some, otherwise we need to rewrite a lot... 

The hooks for ckeditor now serve different purposes or are just triggered once...